contract-examples/test/ExampleFeeManager.ts

```diff
@@ -0,0 +1,229 @@
+    contract = await FeeManager.deploy()
+    await contract.deployed()
+    const contractAddress: string = contract.address
+    console.log(`Contract deployed to: ${contractAddress}`)
+
+    const signers: SignerWithAddress[] = await ethers.getSigners()
+    manager = signers.slice(-1)[0]
+    nonEnabled = signers.slice(-2)[0]
+
+    const feeManager = await ethers.getContractAt("IFeeManager", FEE_MANAGER, owner);
+    let tx = await feeManager.setEnabled(manager.address);
+    await tx.wait()
+
+    tx = await owner.sendTransaction({
+      to: manager.address,
+      value: ethers.utils.parseEther("1")
+    })
+    await tx.wait()
+
+    tx = await owner.sendTransaction({
+      to: nonEnabled.address,
+      value: ethers.utils.parseEther("1")
+    })
+    await tx.wait()
+  });
+
+  it("should add contract deployer as owner", async function () {
+    const contractOwnerAddr: string = await contract.owner()
+    expect(owner.address).to.equal(contractOwnerAddr)
+  });
+
+  // this contract is not given minter permission yet, so should not mintdraw
+  it("contract should not be able to change fee without enabled", async function () {
+    const managerPrecompile = await ethers.getContractAt("IFeeManager", FEE_MANAGER, owner);
+    let contractRole = await managerPrecompile.readAllowList(contract.address);
+    expect(contractRole).to.be.equal(ROLES.NONE)
+    try {
+      let tx = await contract.enableWAGMIFees()
+      await tx.wait()
+    }
+    catch (err) {
+      return
+    }
+    expect.fail("should have errored")
+  })
+
+  it("contract should be added to manager list", async function () {
+    const feeManager = await ethers.getContractAt("IFeeManager", FEE_MANAGER, owner);
+    let adminRole = await feeManager.readAllowList(adminAddress);
+    expect(adminRole).to.be.equal(ROLES.ADMIN)
+    let contractRole = await feeManager.readAllowList(contract.address);
+    expect(contractRole).to.be.equal(ROLES.NONE)
+
+    let enableTx = await feeManager.setEnabled(contract.address);
+    await enableTx.wait()
+    contractRole = await feeManager.readAllowList(contract.address);
+    expect(contractRole).to.be.equal(ROLES.ENABLED)
+  });
+
+  it("admin should be able to change fees through contract", async function () {
+    let enableTx = await contract.enableCustomFees(LOW_FEES)
+    let txRes = await enableTx.wait()
+
+    var res = await contract.getCurrentFeeConfig()
+    expect(res.gasLimit).to.equal(LOW_FEES.gasLimit)
+    expect(res.minBaseFee).to.be.equal(LOW_FEES.minBaseFee)
+
+    var res = await contract.getFeeConfigLastChangedAt()
+
+    expect(res).to.equal(txRes.blockNumber)
+  })
+
+  it("should let low fee tx to be in mempool", async function () {
+    var res = await contract.getCurrentFeeConfig()
+    expect(res.minBaseFee).to.be.equal(LOW_FEES.minBaseFee)
+
+    var testMaxFeePerGas = HIGH_FEES.minBaseFee - 10000
```

Comment:
It's not clear how this fee relates to the set fee because you grab constants from both HIGH and LOW fees. I'm not sure which is greater. It's hard to judge what this test is doing.

Reply:
right!